lib/Model.ts

```diff
@@ -692,6 +693,52 @@ export class Model<TDocument extends { _id?: any }, TInstance> {
         });
     }
 
+    private _mapReduce<Key, Value>(instanceType: InstanceImplementation<MapReducedDocument<Key, Value>, any>, options: MongoDB.MapReduceOptions): Bluebird<void> {
+        return new Bluebird<void>((resolve, reject) => {
+            if (instanceType.mapReduceOptions == undefined)
+                reject(new Error("Cannot mapReduce to this instance."))
+            else
+                this.collection.mapReduce(instanceType.mapReduceOptions.map, instanceType.mapReduceOptions.reduce, options, (err, data) => {
+                    if (err) return reject(err)
+                    return resolve()
+                })
+        })
+    }
+
+    /**
+     * 
```

Comment:
In the inline version where the first parameter is an object with map and reduce function, it can work without any `option.out.inline = 1` being passed, cause it treats it as inline by default. If `option` is constrained as you described, this default behaviour won't be possible.